lib/HookCodeFactory.js

```diff
@@ -55,18 +55,24 @@ class HookCodeFactory {
 				});
 				let code = "";
 				code += '"use strict";\n';
-				code += "return new Promise((_resolve, _reject) => {\n";
+				// Extract _this
+				code += "var _this = this;\n";
+				code += "return new Promise(function(_resolve, _reject) {\n";
 				if (errorHelperUsed) {
 					code += "var _sync = true;\n";
 					code += "function _error(_err) {\n";
 					code += "if(_sync)\n";
-					code += "_resolve(Promise.resolve().then(() => { throw _err; }));\n";
+					code +=
+						"_resolve(Promise.resolve().then(function() { throw _err; }));\n";
 					code += "else\n";
 					code += "_reject(_err);\n";
 					code += "};\n";
 				}
-				code += this.header();
-				code += content;
+
+				// Replace use of 'this' with previously extracted '_this'
+				// in header and in content
+				code += this.header().replace(/this\./g, "_this.");
+				code += content.replace(/this\./g, "_this.");
```

Comment:
Bad idea

Reply:
@evilebottnawi any alternate solution?

Reply:
@evilebottnawi As code is generated with string, why do you think it would be a problem?
Also, do you have a better solution for above situation? If so, can you please share it?  I will work on it asap.